### ISSUE TITLE
"Error during WebSocket handshake: Unexpected response code: 413" error in Chrome

### ISSUE THREAD

**Issue by nhouse** [voryx/Thruway#89]
I'm using AutobahnJS + Thruway on a Windows development machine running Wampserver. Everything has been working fine for quite a while until today, when in Chrome I started seeing this error in the console:

> WebSocket connection to 'ws://mysite.com:9090/' failed: Error during WebSocket handshake: Unexpected response code: 413

I don't get the error in IE or FF, only Chrome.

There is no output from SimpleWsServer.php when AutobahnJS tries to connect from Chrome.

I haven't changed any code related to Thruway or AutobahnJS, but I believe Chrome *may* have updated today to `42.0.2311.90 m`.

The only thing I've tried thus far is restarting my computer a few times, but no change.

Has anyone else experienced this? Any suggestions on how to debug the problem?

**Comment by mbonneau** [comment]
I was not able to reproduce it, I have the same version of chrome (64-bit Mac though). I tried using SimpleWsRouter.php with the plunkr in the readme: http://plnkr.co/edit/8vcBDUzIhp48JtuTGIaj?p=info 

The only place I can see a 413 coming back to you (if you are connecting directly to the router) is line 45 of HttpServer:

https://github.com/ratchetphp/Ratchet/blob/master/src/Ratchet/Http/HttpServer.php#L45

This would be caused if the Request header is larger than 4k, which doesn't seem likely. You could test it and see what happens if you increase the buffer size here:

https://github.com/ratchetphp/Ratchet/blob/master/src/Ratchet/Http/HttpRequestParser.php#L20

Try 8k maybe.

Another place you can look is in the network tab of the debug window in Chrome. See if the header looks abnormal.

Please let me know if this is the issue. I will dig deeper if this doesn't locate the issue.

**Comment by nhouse** [comment]
Thanks so much for your help, @mbonneau!

I didn't think to review the request header, but upon inspecting it in Chrome I noticed the cookie string seemed much larger than usual. I pasted it into a text file and saved it, and the resulting file size was 3.91KB.

I cleared the cookies and the issue went away, so I'm guessing your analysis was correct.

I don't know what caused there to be so many cookies, but I may try increasing the 4k limit to 8k as you suggested just in case this happens again somehow (again, I'm pretty sure Chrome updated, so maybe something weird happened with cookies at that time).

Thanks again for your help!
